### PR TITLE
Remove score multiplier on difficulty-increasing mods with customised settings

### LINKED ISSUE #7698: Score multiplier remains the same on speed-adjusting mods even when reduced
**Describe the bug:** 
Not entirely sure if you could call this a bug, but this really shouldn't happen...
Whenever you enable a speed-adjusting mod, (Double Time, Nightcore, Half Time, Daycore) you get options to customize the amount it speeds up / slows down the beatmap.
However, no matter how much you speed up / slow down the beatmap, the score multiplier remains the same - so if you were to play a map with Nightcore, but sped up by only 1.01x, you would have the same multiplier as someone playing with 2.0x speed. (multiplier for both is 1.12x.)

**Screenshots or videos showing encountered issue:** 
https://youtu.be/tJbYDlbXIGE 

**osu!lazer version:** 
2020.130.0

**Logs:** 
uhhhhh

### PR DESCRIPTION
**Pull request by frenzibyte**
- Closes #7698 

As mentioned in https://github.com/ppy/osu/discussions/19191#discussioncomment-3166121, as a first step, all mods with increased score multiplier are nullified if the mod uses custom configuration.

I've added a property for this to allow including it in all mods which have an increased score multiplier, to avoid requiring to add every new setting in the multiplier field (e.g. `OsuModFlashlight` which has three different settings with varying states to abuse it).

I've considered moving this to a higher level but realised it'll have to be specific to mods with a >1 score multiplier, and also somehow still allow mods which have a proper customisation-respecting score multiplier (in the future), therefore I went with doing it on a per-mod basis instead.

### CODE REVIEW

**Comment by peppy** [comment]
An alternative implementation would be to apply this globally to all mods. And potentially set the multiplier to something very low by default until we address the larger issue (ie. to handle people using difficulty adjust and changing the settings to make a beatmap much easier).

Will wait to hear @smoogipoo (and anyone else that cares to chime in) thoughts on this one.

**Comment by Moltanicaa** [comment]
I have a couple of thoughts on this.  
1. As far as DA is concerned, the biggest problem is changing the size of the notes. When the size is changed, the starrate changes, so the change in the multiplier can be tied to the ratio of the change in the starrate from the usual one. Same 
2. As for the DT/HT customization, even a 1.01x multiplier still speeds up the map by 1%, so it would be fair to also have a dynamic multiplier, so instead of 1.12x one could give...1.024x (1+0.12/50, so how do we have 50 values ​​from the usual 1.5x speed)? This would give me a total bonus of +2400 points (if i will SS the map), which is not much.

**Comment by peppy** [comment]
> I have a couple of thoughts on this.  
> 1. As far as DA is concerned, the biggest problem is changing the size of the notes. When the size is changed, the starrate changes, so the change in the multiplier can be tied to the ratio of the change in the starrate from the usual one.  
> 2. As for the DT/HT customization, even a 1.01x multiplier still speeds up the map by 1%, so it would be fair to also have a dynamic multiplier, so instead of 1.12x one could give...1.024x (1+0.12/50, so how do we have 50 values ​​from the usual 1.5x speed)? This would give me a total bonus of +2400 points (if i will SS the map), which is not much.

Both valid points but also for a different discussion as I mentioned. This is an initial fix until we figure things out.

File: osu.Game/Overlays/Mods/ModSelectOverlay.cs [hunk 1]
```diff
@@ -238,9 +240,14 @@ protected override void LoadComplete()
...
             SelectedMods.BindValueChanged(val =>
             {
+                modSettingChangeTracker?.Dispose();
+
                 updateMultiplier();
                 updateCustomisation(val);
                 updateFromExternalSelection();
+
+                modSettingChangeTracker = new ModSettingChangeTracker(val.NewValue);
+                modSettingChangeTracker.SettingChanged += _ => updateMultiplier();
```
Comment (smoogipoo) [inline osu.Game/Overlays/Mods/ModSelectOverlay.cs:250]: Should probably be around an `if (AllowCustomisation)` similar to the above optimisation. Concern is selecting all mods in the freemods overlay.

**Review by smoogipoo — APPROVED** [review]
I think this is probably fine for the time being.

**Review by smoogipoo — CHANGES_REQUESTED** [review]
Uh.. Minus the review above

**Review by peppy — APPROVED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**